python/cudf/cudf/core/column/string.py

```diff
@@ -2105,47 +2158,93 @@ def rsplit(self, pat=None, n=-1, expand=None, **kwargs):
         found splits:
 
             - If found splits > n, make first n splits only
             - If found splits <= n, make all splits
             - If for a certain row the number of found splits < n,
-              append None for padding up to n.
+              append None for padding up to n if ``expand=True``.
+
+        If using ``expand=True``, Series and Index callers return
+        DataFrame and MultiIndex objects, respectively.
 
         Examples
         --------
         >>> import cudf
-        >>> data = ["this is a regular sentence",
-        ... "https://docs.python.org/3/tutorial/index.html",
-        ... None]
-        >>> s = cudf.Series(data)
+        >>> s = cudf.Series(
+        ...     [
+        ...         "this is a regular sentence",
+        ...         "https://docs.python.org/3/tutorial/index.html",
+        ...         None
+        ...     ]
+        ... )
+        >>> s
+        0                       this is a regular sentence
+        1    https://docs.python.org/3/tutorial/index.html
+        2                                             <NA>
+        dtype: object
+
+        In the default setting, the string is split by whitespace.
+
+        >>> s.str.rsplit()
+        0                   [this, is, a, regular, sentence]
+        1    [https://docs.python.org/3/tutorial/index.html]
+        2                                               None
+        dtype: list
+
+        Without the ``n`` parameter, the outputs of ``rsplit``
+        and ``split`` are identical.
+
+        >>> s.str.split()
+        0                   [this, is, a, regular, sentence]
+        1    [https://docs.python.org/3/tutorial/index.html]
+        2                                               None
+        dtype: list
+
+        The n parameter can be used to limit the number of
+        splits on the delimiter. The outputs of split and rsplit are different.
+
         >>> s.str.rsplit(n=2)
+        0                     [this is a, regular, sentence]
+        1    [https://docs.python.org/3/tutorial/index.html]
+        2                                               None
+        dtype: list
+        >>> s.str.split(n=2)
+        0                     [this, is, a regular sentence]
+        1    [https://docs.python.org/3/tutorial/index.html]
+        2                                               None
+        dtype: list
+
+        When using ``expand=True``, the split elements will expand
+        out into separate columns. If ``<NA>`` value is present,
+        it is propagated throughout the columns during the split.
+
+        >>> s.str.rsplit(n=2, expand=True)
                                                        0        1         2
         0                                      this is a  regular  sentence
-        1  https://docs.python.org/3/tutorial/index.html     None      None
-        2                                           None     None      None
```

Comment:
I think string docs will need an update where `None` exists. Our new convention to represent nulls is `<NA>`. So will tackle it in a separate PR.